app/assets/javascripts/components/OnPageFeedback.js

```diff
@@ -89,15 +102,56 @@ define(['jquery', 'DoughBaseComponent', 'common'], function($, DoughBaseComponen
   OnPageFeedback.prototype._updateCount = function(data) {
     var countResponse = data;
 
-    this.likeCount.text(countResponse.likes_count);
-    this.dislikeCount.text(countResponse.dislikes_count);
+    this.likeElement.text(countResponse.likes_count);
+    this.dislikeElement.text(countResponse.dislikes_count);
   };
 
   OnPageFeedback.prototype._showPage = function(pageName) {
     this.pages.addClass('is-hidden');
     this.pages.filter('[data-dough-feedback-page=' + pageName + ']').removeClass('is-hidden');
   };
 
+  OnPageFeedback.prototype._restoreState = function() {
+    var data = localStorage['MAS.onPageFeedback.' + this.pageId];
+
+    if (typeof(data) != 'undefined') {
+      var json = JSON.parse(data)
+
+      // Ignore stored data if more than 30 days old
+      if (json.time > (Date.now() - (30*24*60*60*1000))) {
+        this.likeElement.text(json.likes_count);
+        this.dislikeElement.text(json.dislikes_count);
+        this.state = json.last_state;
+      }
+    }
+  };
+
+  OnPageFeedback.prototype._storeState = function() {
+    var data = {
+      time: Date.now(),
+      likes_count: this.likeElement.text(),
+      dislikes_count: this.dislikeElement.text(),
+      last_state: this.state
+    }
+
+    localStorage['MAS.onPageFeedback.' + this.pageId] = JSON.stringify(data);
+  };
+
+  OnPageFeedback.prototype._applyState = function() {
```

Comment:
After discussion with @jongilbraith, this doesn't quite work, but we are going to have a think about the naming involved in this component. But that's not part of this PR.